packages/react/src/component-tests/Form/Form.cy.tsx

```diff
@@ -0,0 +1,258 @@
+const CHECKBOX = '[type="checkbox"]';
+const RADIO = '[type="radio"]';
+const DAY_INPUT = ".day-input";
+const MONTH_INPUT = ".month-input";
+const YEAR_INPUT = ".year-input";
+// Keyboard keys
+const ESCAPE_KEY = "Escape";
+const ARROW_DOWN_KEY = "ArrowDown";
+const TAB_KEY = "Tab";
+const SPACE_KEY = "Space";
+// Other
+const CONSOLE_LOG = "@consoleLog";
+const DATE_VAL = "val";
+const STEP_TYPE = "step-type";
+const CURRENT = "current";
+const CHECKED = "checked";
+const SELECTED = "selected";
+const VALIDATION_STATUS = "validation-status";
+const ERROR = "error";
+
+//* METHODS
+const checkCurrentStep = (step: number) => {
+  cy.get(IC_STEP)
+    .eq(step)
+    .should(BE_VISIBLE)
+    .should(HAVE_ATTR, STEP_TYPE, CURRENT);
+};
+
+const findShadowEl = (
+  element: string,
+  selector: string
+): Cypress.Chainable<JQuery<HTMLElement>> =>
+  cy.get(`${element}`).shadow().find(`${selector}`);
+
+const clickOnShadowEl = (
+  element: string,
+  selector: string,
+  order: number = 0
+): void => {
+  cy.get(element).shadow().find(selector)?.eq(order).click();
+};
+
+const checkDateInputValue = (date: Date | null) => {
+  const currDay = (date && date.getDate()) || 0;
+  const currMon = (date && date.getMonth() + 1) || 0;
+  const currYear = (date && date.getFullYear()) || 0;
+  let dayVal: number;
+  let monthVal: number;
+  let yearVal: number;
+  cy.findShadowEl(IC_DATE_PICKER, IC_DATE_INPUT)
+    .shadow()
+    .find(DAY_INPUT)
+    .invoke(DATE_VAL)
+    .then((val: string) => {
+      dayVal = Number(val);
+    })
+    .then(() => {
+      expect(dayVal).to.eq(currDay);
+    });
+  cy.findShadowEl(IC_DATE_PICKER, IC_DATE_INPUT)
+    .shadow()
+    .find(MONTH_INPUT)
+    .invoke(DATE_VAL)
+    .then((val: string) => {
+      monthVal = Number(val);
+    })
+    .then(() => {
+      expect(monthVal).to.eq(currMon);
+    });
+  cy.findShadowEl(IC_DATE_PICKER, IC_DATE_INPUT)
+    .shadow()
+    .find(YEAR_INPUT)
+    .invoke(DATE_VAL)
+    .then((val: string) => {
+      yearVal = Number(val);
+    })
+    .then(() => {
+      expect(yearVal).to.eq(currYear);
+    });
+};
+
+describe("Coffee subscription form", () => {
+  beforeEach(() => {
+    cy.viewport("macbook-16");
+    cy.window().then((win) => {
+      cy.spy(win.console, "log").as("consoleLog");
+    });
+  });
+  it("should fill the form with no errors/validation", () => {
+    mount(<Form />);
+    // checkHydrated will wait until the component is hydrated and ensures it is ready to test
+    cy.checkHydrated(IC_PAGE_HEADER);
+
+    // Check the first step is visible
+    cy.get(IC_STEPPER).should(BE_VISIBLE);
+    checkCurrentStep(0);
+
+    // Select a radio option
+    cy.findShadowEl(IC_RADIO_OPTION, RADIO).first().check({ force: true });
+    cy.get(IC_RADIO_OPTION).eq(0).should(HAVE_ATTR, SELECTED);
+
+    // Select a select option using mouse
+    clickOnShadowEl(IC_SELECT, IC_INPUT_CONTAINER);
+    clickOnShadowEl(IC_SELECT, IC_MENU_OPTION, 3);
```

Comment:
```suggestion
    clickOnShadowEl(IC_SELECT, IC_MENU_OPTION+'[data-value="aeropress"]');
```
Same reasoning as above, selecting by value rather than number in the list.